src/modules/sensors/vehicle_angular_velocity/VehicleAngularVelocity.cpp

```diff
@@ -0,0 +1,212 @@
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "VehicleAngularVelocity.hpp"
+
+#include <px4_log.h>
+
+using namespace time_literals;
+using namespace matrix;
+
+VehicleAngularVelocity::VehicleAngularVelocity() :
+	ModuleParams(nullptr),
+	WorkItem(px4::wq_configurations::rate_ctrl),
+	_cycle_perf(perf_alloc(PC_ELAPSED, "vehicle_angular_velocity: cycle time")),
+	_interval_perf(perf_alloc(PC_INTERVAL, "vehicle_angular_velocity: interval")),
+	_sensor_gyro_latency_perf(perf_alloc(PC_ELAPSED, "vehicle_angular_velocity: sensor gyro latency"))
+{
+	_gyro_count = math::constrain(orb_group_count(ORB_ID(sensor_gyro)), 1, MAX_GYRO_COUNT);
+
+	/* initialize thermal corrections as we might not immediately get a topic update (only non-zero values) */
+	for (unsigned i = 0; i < 3; i++) {
+		// used scale factors to unity
+		_sensor_correction.gyro_scale_0[i] = 1.0f;
+		_sensor_correction.gyro_scale_1[i] = 1.0f;
+		_sensor_correction.gyro_scale_2[i] = 1.0f;
+	}
+
+	parameters_updated();
+}
+
+VehicleAngularVelocity::~VehicleAngularVelocity()
+{
+	stop();
+
+	perf_free(_cycle_perf);
+	perf_free(_interval_perf);
+	perf_free(_sensor_gyro_latency_perf);
+}
+
+void
+VehicleAngularVelocity::start()
+{
+	_sensor_correction_sub.register_callback();
+
+	selected_gyro_update();
+}
+
+void
+VehicleAngularVelocity::stop()
+{
+	Deinit();
+
+	// clear all registered callbacks
+	for (auto sub : _sensor_gyro_sub) {
+		sub.unregister_callback();
+	}
+}
+
+void
+VehicleAngularVelocity::parameters_updated()
+{
+	/* get transformation matrix from sensor/board to body frame */
+	_board_rotation = get_rot_matrix((enum Rotation)_param_sens_board_rot.get());
+
+	/* fine tune the rotation */
+	const Dcmf board_rotation_offset(Eulerf(
+			math::radians(_param_sens_board_x_off.get()),
+			math::radians(_param_sens_board_y_off.get()),
+			math::radians(_param_sens_board_z_off.get())));
+
+	_board_rotation = board_rotation_offset * _board_rotation;
+}
+
+void
+VehicleAngularVelocity::parameter_update_poll()
+{
+	/* Check if parameters have changed */
+	parameter_update_s param_update;
+
+	if (_params_sub.update(&param_update)) {
+		updateParams();
+		parameters_updated();
+	}
+}
+
+bool
+VehicleAngularVelocity::selected_gyro_update()
+{
+	// check if the selected gyro has updated first
+	_sensor_correction_sub.update(&_sensor_correction);
+
+	/* update the latest gyro selection */
+	if (_selected_gyro != _sensor_correction.selected_gyro_instance) {
+		if (_sensor_correction.selected_gyro_instance < _gyro_count) {
```

Comment:
wrap in
```
if (_sensor_correction.selected_gyro_instance < MAX_GYRO_COUNT) {
}
```
and issue an error if the selected gyro instance number is too large

Reply:
ok my bad, `_gyro_count` is always lower than MAX_GYRO_COUNT.

Reply:
This is a transplant from the legacy code I think we can drop entirely.